.github/workflows/workflow-java.yaml

```diff
@@ -62,15 +36,20 @@ jobs:
           echo "Is github.event_name != 'pull_request': ${{ github.event_name != 'pull_request' }}"
           echo "=== Check environment conditionals"
           echo "inputs.do_staging_build: ${{ inputs.do_staging_build }}"
-          echo "inputs.do_production_build: ${{ inputs.do_production_build }}"
   # Test projects
   test_mvn:
+    # If we are in a commit from the [actions] pipeline (see java-java-update-version.yaml), do nothing
+    if: |
+      !contains(github.event.head_commit.message, '[actions]')
     name: "Run tests via mvn"
     uses: ./.github/workflows/run_mvn.yaml
     with:
       script: "test --no-transfer-progress"
       publish_surefire_test_report: true
   build_mvn:
+    # If we are in a commit from the [actions] pipeline (see java-java-update-version.yaml), do nothing
+    if: |
+      !contains(github.event.head_commit.message, '[actions]')
```

Comment:
I don't think this condition will ever be satisfied for the reason you linked to on Jira.

Also, I think this boolean trap arose because the logic was placed too low in the stack. It's trying to guess what the caller wants, without having all the information.